index.ts

```diff
@@ -81,160 +104,102 @@ class Analytics {
     });
   }
 
-  _validate(message, type) {
-    try {
-      looselyValidate(message, type);
-    } catch (e) {
-      if (e.message === 'Your message must be < 32kb.') {
-        this.logger.info(
-          'Your message must be < 32kb. This is currently surfaced as a warning. Please update your code',
-          message
-        );
-        return;
-      }
-      throw e;
-    }
-  }
-
   /**
-   * Send an identify `message`.
-   *
-   * @param {Object} message
-   * @param {String=} message.userId (optional)
-   * @param {String=} message.anonymousId (optional)
-   * @param {Object=} message.context (optional)
-   * @param {Object=} message.traits (optional)
-   * @param {Object=} message.integrations (optional)
-   * @param {Date=} message.timestamp (optional)
-   * @param {Function=} callback (optional)
-   * @return {Analytics}
+   * Sends an "identify" message that associates traits with a user.
    */
-
-  identify(message, callback) {
-    this._validate(message, 'identify');
+  identify(message: AnalyticsMessage, callback?: AnalyticsMessageCallback): Analytics {
+    this.validate(message, 'identify');
```

Comment:
nit: prefer `AnalyticsEventType` over `string` for internals. 

but i don't feel too strongly